packages/client/src/schema/filters.ts

```diff
@@ -114,12 +115,27 @@ export type Filter<T> = T extends Record<string, any>
     : BaseApiFilter<T> | NestedApiFilter<T>
   : PropertyFilter<T>;
 
-export function cleanFilter(filter?: FilterExpression) {
-  if (!filter) return undefined;
+export function cleanFilter(filter?: FilterExpression | FilterPredicate): FilterExpression | undefined {
+  if (!isDefined(filter)) return undefined;
 
-  const values = Object.values(filter)
-    .filter(Boolean)
-    .filter((value) => (Array.isArray(value) ? value.length > 0 : true));
+  const values = Object.fromEntries(
+    Object.entries(filter).reduce((acc, [key, value]) => {
+      // Remove null and undefined values
+      if (!isDefined(value)) return acc;
 
-  return values.length > 0 ? filter : undefined;
+      if (Array.isArray(value)) {
```

Comment:
Do we have a test for this `if` branch of code? 

I can't see how the test you added covers it, but perhaps there is a prexisting one?

I think we should have a test the `if (clean.length === 0) return acc;` case and the `return [...acc, [key, clean]];` case?